nautobot_circuit_maintenance/handle_notifications/handler.py

```diff
@@ -167,59 +167,43 @@ def process_parsed_notification(
-    """
-    provider = Provider.objects.filter(slug=notification.provider_type).last()
+def get_maintenances_from_notification(logger: Job, notification: MaintenanceNotification, provider: Provider):
+    """Use the `circuit_maintenance_parser` library to get Maintenances from the notification."""
+    provider_type = provider.cf.get("provider_parser_circuit_maintenances", "").lower() or provider.slug
 
-    if not provider:
-        logger.log_warning(
-            message=(
-                f"Raw notification not created because is referencing to a provider not existent: {notification.provider_type}"
-            )
-        )
+    parser_provider = init_provider(provider_type=provider_type)
+    if not parser_provider:
+        logger.log_warning(message=f"Notification Parser not found for {notification.provider_type}")
         return None
 
-    provider_type = provider.cf.get("provider_parser_circuit_maintenances", "").lower() or provider.slug
+    # TODO: we should handle NotificationData initialization
+    data_to_process = init_data_email(notification.raw_payload)
 
-    raw_payload = b""
-    for raw_payload in notification.raw_payloads:
-        parser = init_provider(raw=raw_payload, provider_type=provider_type)
-        if not parser:
-            logger.log_warning(message=f"Notification Parser not found for {notification.provider_type}")
-            return None
+    try:
+        return parser_provider.get_maintenances(data_to_process)
+    except ProviderError:
+        tb_str = traceback.format_exc()
+        logger.log_info(message=f"Parsing failed for notification `{notification.subject}`:\n```\n{tb_str}\n```")
+        return None
+    except Exception:
+        tb_str = traceback.format_exc()
+        logger.log_info(
+            message=f"Unexpected exception while parsing notification `{notification.subject}`.\n```\n{tb_str}\n```"
+        )
+        return None
 
-        try:
-            parsed_notifications = parser.process()
-            break
-        except ParsingError:
-            tb_str = traceback.format_exc()
-            logger.log_info(message=f"Parsing failed for notification `{notification.subject}`:\n```\n{tb_str}\n```")
-        except Exception:
-            tb_str = traceback.format_exc()
-            logger.log_info(
-                message=f"Unexpected exception while parsing notification `{notification.subject}`.\n```\n{tb_str}\n```"
-            )
-    else:
-        parsed_notifications = []
-        logger.log_warning(message=f"Parsed failed for all the raw payloads for `{notification.subject}`.")
-        # Carry on with the last raw_payload in the list, if any
-        if not raw_payload:
-            return None
 
-    if isinstance(raw_payload, str):
-        raw_payload = raw_payload.encode("utf-8")
+def create_raw_nofitication(logger: Job, notification: MaintenanceNotification, provider: Provider):
```

Comment:
```suggestion
def create_raw_notification(logger: Job, notification: MaintenanceNotification, provider: Provider):
```

Reply:
it took me 30 seconds to spot the typo 😄